src/test/java/org/jenkinsci/plugins/envinject/BuildCauseRetrieverTest.java

```diff
@@ -139,6 +142,32 @@ public String getShortDescription() {
         assertThat(build, withCause(ROOT_BUILD_CAUSE, ""));
     }
 
+    @SuppressWarnings("deprecation")
+    @Test
+    public void shouldWriteInfoUserCause() throws Exception {
+    	UserCause cause = Cause.UserCause.class.newInstance();
+        FreeStyleBuild build = jenkins.createFreeStyleProject().scheduleBuild2(0, cause).get();
+
+        assertThat(build.getResult(), is(SUCCESS));
+        assertThat(build, withCause(BUILD_CAUSE, MANUAL_TRIGGER));
+        assertThat(build, withCause(ROOT_BUILD_CAUSE, MANUAL_TRIGGER));
+        assertThat(build, withCausesIsTrue(sub(BUILD_CAUSE, MANUAL_TRIGGER), sub(ROOT_BUILD_CAUSE, MANUAL_TRIGGER)));
+        assertThat(build, withCause(USER_NAME, cause.getUserName()));
+    }
+
+    @Test
+    public void shouldWriteInfoUserIdCause() throws Exception {
+    	UserIdCause cause = Cause.UserIdCause.class.newInstance();
```

Comment:
and where user_id is set to check it? You just copied prev testcase. Unit test should test only what you want to check. Please add separate cases for nonnull/null user_id and user_name only. Don't touch other cases if they works fine.